lib/active_fedora/nested_attributes.rb

```diff
@@ -148,13 +149,14 @@ def assign_nested_attributes_for_collection_association(association_name, attrib
         attributes = attributes.with_indifferent_access
 
         if attributes['id'].blank?
-          association.build(attributes.except(*UNASSIGNABLE_KEYS)) unless call_reject_if(association_name, attributes)
-
+          if !has_destroy_flag?(attributes) || !allow_destroy
```

Comment:
Can't say I'm entirely familiar with this either. But my reasoning was that the association.build line has an unless clause already so I assumed it'd be safe to just skip that line if for whatever reason we don't want to create the association.